src/vast_tracker.js

```diff
@@ -368,84 +376,125 @@ export class VASTTracker extends EventEmitter {
    * additional portion of the ad.
    * Calls the acceptInvitation tracking URLs.
    *
+   * @param {Object} [macros={}] - An optional Object containing macros and their values to be used and replaced in the tracking calls.
    * @emits VASTTracker#acceptInvitation
    */
-  acceptInvitation() {
-    this.track('acceptInvitation');
+  acceptInvitation(macros = {}) {
+    this.track('acceptInvitation', macros);
   }
 
   /**
    * Must be called if user activated a control to expand the creative.
    * Calls the adExpand tracking URLs.
    *
+   * @param {Object} [macros={}] - An optional Object containing macros and their values to be used and replaced in the tracking calls.
    * @emits VASTTracker#adExpand
    */
-  adExpand() {
-    this.track('adExpand');
+  adExpand(macros = {}) {
+    this.track('adExpand', macros);
   }
 
   /**
    * Must be called when the user activated a control to reduce the creative to its original dimensions.
    * Calls the adCollapse tracking URLs.
    *
+   * @param {Object} [macros={}] - An optional Object containing macros and their values to be used and replaced in the tracking calls.
    * @emits VASTTracker#adCollapse
    */
-  adCollapse() {
-    this.track('adCollapse');
+  adCollapse(macros = {}) {
+    this.track('adCollapse', macros);
   }
 
   /**
    * Must be called if the user clicked or otherwise activated a control used to minimize the ad.
    * Calls the minimize tracking URLs.
+   *
+   * @param {Object} [macros={}] - An optional Object containing macros and their values to be used and replaced in the tracking calls.
    * @emits VASTTracker#minimize
    */
-  minimize() {
-    this.track('minimize');
+  minimize(macros = {}) {
+    this.track('minimize', macros);
+  }
+
+  /**
+   * Must be called if the player did not or was not able to execute the provided
+   * verification code.The [REASON] macro must be filled with reason code
+   * Calls the verificationNotExecuted trackings URLs.
+   *
+   * @param {Object} [macros={}] - An optional Object containing macros and their values to be used and replaced in the tracking calls.
+   * @emits VASTTracker#verificationNotExecuted
+   */
+  verificationNotExecuted(macros = {}) {
+    if (
+      !(this.ad && this.ad.adVerifications && this.ad.adVerifications.length)
```

Comment:
this is a bit hard to read, we could just write:
```
!this.ad || !this.ad.adVerifications || !this.ad.adVerifications.length
```

Reply:
but I'm not sure we should do this check. IMO this should break if you try to call it when there is no `adVerifications`. Or we should send an error. @rumesh what do you think?

Reply:
Y6es I agree, this function shouldn't be called if no adVerifications, but to be clean and robuste vast-client should send an error saying no adVerification.
Because if we do not have this check it will be an error like it wasn't handle properly by the vast-client.
So agree on this should break but not because accessing `this.ad.adVerifications` while it's not defined but because we throw an error in this case

Reply:
Done !